tests/eibc_invariant_test.go

```diff
@@ -0,0 +1,750 @@
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	err = ic.Build(ctx, eRep, test.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: true,
+
+		// This can be used to write to the block database which will index all block data e.g. txs, msgs, events, etc.
+		// BlockDatabaseFile: test.DefaultBlockDatabaseFilepath(),
+	})
+	require.NoError(t, err)
+
+	CreateChannel(ctx, t, r1, eRep, dymension.CosmosChain, rollapp1.CosmosChain, ibcPath)
+	CreateChannel(ctx, t, r2, eRep, dymension.CosmosChain, rollapp2.CosmosChain, anotherIbcPath)
+
+	walletAmount := math.NewInt(1_000_000_000_000)
+
+	// Create some user accounts on both chains
+	users := test.GetAndFundTestUsers(t, ctx, t.Name(), walletAmount, dymension, dymension, rollapp1)
+
+	// Wait a few blocks for relayer to start and for user accounts to be created
+	err = testutil.WaitForBlocks(ctx, 5, dymension, rollapp1)
+	require.NoError(t, err)
+
+	// Get our Bech32 encoded user addresses
+	dymensionUser, marketMaker, rollappUser := users[0], users[1], users[2]
+
+	dymensionUserAddr := dymensionUser.FormattedAddress()
+	marketMakerAddr := marketMaker.FormattedAddress()
+	rollappUserAddr := rollappUser.FormattedAddress()
+
+	// Assert the accounts were funded
+	testutil.AssertBalance(t, ctx, dymension, dymensionUserAddr, dymension.Config().Denom, walletAmount)
+	testutil.AssertBalance(t, ctx, dymension, marketMakerAddr, dymension.Config().Denom, walletAmount)
+	testutil.AssertBalance(t, ctx, rollapp1, rollappUserAddr, rollapp1.Config().Denom, walletAmount)
+
+	transferAmount := math.NewInt(1_000_000)
+	multiplier := math.NewInt(10)
+
+	eibcFee := transferAmount.Quo(multiplier) // transferAmount * 0.1
+	transferAmountWithoutFee := transferAmount.Sub(eibcFee)
+
+	dymChannels, err := r1.GetChannels(ctx, eRep, dymension.Config().ChainID)
+	require.NoError(t, err)
+
+	channsRollApp1, err := r1.GetChannels(ctx, eRep, rollapp1.GetChainID())
+	require.NoError(t, err)
+
+	channsRollApp2, err := r2.GetChannels(ctx, eRep, rollapp2.GetChainID())
+	require.NoError(t, err)
+
+	require.Len(t, channsRollApp1, 1)
+	require.Len(t, channsRollApp2, 1)
+	require.Len(t, dymChannels, 2)
+
+	triggerHubGenesisEvent(t, dymension, rollappParam{
+		rollappID: rollapp1.Config().ChainID,
+		channelID: channsRollApp1[0].Counterparty.ChannelID,
+		userKey:   dymensionUser.KeyName(),
+	})
+	triggerHubGenesisEvent(t, dymension, rollappParam{
+		rollappID: rollapp2.Config().ChainID,
+		channelID: channsRollApp2[0].Counterparty.ChannelID,
+		userKey:   dymensionUser.KeyName(),
+	})
+
+	// Start relayer
+	err = r1.StartRelayer(ctx, eRep, ibcPath)
+	require.NoError(t, err)
+	err = r2.StartRelayer(ctx, eRep, anotherIbcPath)
+	require.NoError(t, err)
+
+	epochIdentifier := json.RawMessage(fmt.Sprintf(`"%s"`, "minute"))
+	propTx, err := dymension.ParamChangeProposal(ctx, dymensionUser.KeyName(), &utils.ParamChangeProposalJSON{
```

Comment:
could this one be set in genesis KV for short?

Reply:
okay